src/main/java/chess/controller/WebChessControllerAdvice.java

```diff
@@ -9,13 +12,22 @@
 @RestControllerAdvice(annotations = RestController.class)
 public class WebChessControllerAdvice {
 
-    @ExceptionHandler(value = IllegalArgumentException.class)
-    public ResponseEntity<ErrorResponseDto> handleIllegalArgumentException(IllegalArgumentException e) {
-        return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
+    @ExceptionHandler(value = {IllegalArgumentException.class, IllegalStateException.class,
+        NullPointerException.class, NoSuchElementException.class})
+    public ResponseEntity<ErrorResponseDto> handleIllegalException(RuntimeException e) {
+        return ResponseEntity.badRequest()
+            .body(new ErrorResponseDto(e.getMessage()));
     }
 
-    @ExceptionHandler(value = IllegalStateException.class)
-    public ResponseEntity<ErrorResponseDto> handleIllegalStateException(IllegalStateException e) {
-        return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
+    @ExceptionHandler(value = NotFoundException.class)
+    public ResponseEntity<ErrorResponseDto> handleNotFoundException(NotFoundException e) {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND)
+            .body(new ErrorResponseDto("[ERROR] 방 정보를 찾을 수 없습니다."));
+    }
```

Comment:
요건 코딩 스타일이나 팀 컨벤션이라 참고만 해주세요 😊

응답 값에 `방 정보를 찾을 수 없다`고 명시되어 있기에 예외도 더 구체적인 예외를 사용할 수 있을거 같아요. 

```suggestion
    @ExceptionHandler(value = RoomNotFoundException.class)
    public ResponseEntity<ErrorResponseDto> handleNotFoundException(RoomNotFoundException e) {
        return ResponseEntity.status(HttpStatus.NOT_FOUND)
            .body(new ErrorResponseDto("[ERROR] 방 정보를 찾을 수 없습니다."));
    }
```

요러면 추후 더 복잡한 프로젝트에서 여러 객체에 따른 처리를 세부적으로 해줄 수 있겠죠?

Reply:
처음에 Room으로 작성했는데, 확장성을 미리 고려하지 않고 NotFound로 고쳤습니다!

하지만, 앞으로 더 큰 프로젝트를 시도할 수 있으니 미리 예행연습한다고 생각하고 Room으로 변경하는 습관을 들이는게 좋을 것 같습니다.

이런 사소한 것까지도..ㅠ 감사하네요 토니!